CMakeLists.txt

```diff
@@ -1,9 +1,187 @@
 cmake_minimum_required(VERSION 3.10.0)
+
+option (VERONA_DOWNLOAD_LLVM "Download cached version of LLVM" ON)
+
+# This is used to create options that are passed through to the main Verona
+# build
+if (NOT DEFINED VERONA_EXTRA_CMAKE_ARGS)
+  set (VERONA_EXTRA_CMAKE_ARGS)
+endif ()
+macro(verona_option variable message default)
+  option(variable message default)
+  list (APPEND VERONA_EXTRA_CMAKE_ARGS
+    "-D${variable}=${${variable}}"
+  )
+endmacro()
+
+verona_option(ENABLE_ASSERTS "Enable asserts even in release builds" OFF)
+verona_option(RT_TESTS "Including unit tests for the runtime" OFF)
+verona_option(USE_SCHED_STATS "Track scheduler stats" OFF)
+verona_option(USE_ASAN "Use address sanitizer" OFF)
+verona_option(VERONA_CI_BUILD "Disable features not sensible for CI" OFF)
+verona_option(USE_SYSTEMATIC_TESTING "Enable systematic testing in the runtime" OFF)
+verona_option(VERONA_EXPENSIVE_SYSTEMATIC_TESTING "Increase the range of seeds covered by systematic testing" OFF)
+verona_option(USE_CRASH_LOGGING "Enable crash logging in the runtime" OFF)
+
+# This is to trick the CMake into building LLVM before Verona.
+# We use two External Projects, so that the LLVM build can complete and install
+# before we start the Verona one.  This is required as we consume CMake
+# artifacts, which are not available otherwise.
+include(ExternalProject)
+if (NOT DEFINED VERONA_LLVM_LOCATION)
+  project(verona CXX)
+  # Restrict types of builds to ones where we have cached LLVM
+  set(default_build_type "Release")
+  if (MSVC)
+    if (NOT CMAKE_CONFIGURATION_TYPES)
+      if (NOT CMAKE_BUILD_TYPE)
+        set(CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "Choose the types of build." FORCE)
+      else()
+        set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE}" CACHE STRING "Choose the type of build." FORCE)
+      endif()
+      list (APPEND VERONA_EXTRA_CMAKE_ARGS
+        -DCMAKE_CONFIGURATION_TYPES=${CMAKE_CONFIGURATION_TYPES})
+    endif ()
+  else ()
+    if (NOT CMAKE_BUILD_TYPE)
+      message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+      set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
+    endif()
+    if (NOT CMAKE_CONFIGURATION_TYPES)
+      set(CMAKE_CONFIGURATION_TYPES ${CMAKE_BUILD_TYPE} CACHE STRING "Choose the types of build." FORCE)
+    endif ()
+  endif ()
+
+  if (VERONA_DOWNLOAD_LLVM)
+    find_package(Git)
+    execute_process(
+      COMMAND ${GIT_EXECUTABLE} rev-parse --short=11 HEAD
+      WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/external/llvm-project"
+      OUTPUT_VARIABLE LLVM_PACKAGE_GIT_VERSION
+    )
+
+    string(TOLOWER ${CMAKE_SYSTEM_NAME} PLATFORM)
+    if (${PLATFORM} STREQUAL "darwin")
+      set (PLATFORM "macos")
```

Comment:
We could rename the packages to 'darwin', too. Since we're creating entirely new packages, this would be a good time to do it. :)